lib/src/widgets/buttons/text_button.dart

```diff
@@ -0,0 +1,90 @@
+import 'package:basf_flutter_components/basf_flutter_components.dart';
+import 'package:flutter/material.dart';
+
+/// {@template basf_text_button}
+/// A BASF-style text button
+/// {@endtemplate}
+class BasfTextButton extends BasfButton {
+  /// {@macro basf_text_button}
+  /// contained
+  const BasfTextButton.contained({
+    super.key,
+    super.text,
+    super.leadingIcon,
+    super.trailingIcon,
+    super.iconSize,
+    super.child,
+    super.onPressed,
+    super.onLongPress,
+    super.style,
+    super.size,
+    super.expanded,
+  });
+
+  /// {@macro basf_text_button}
+  /// transparent
+  BasfTextButton.transparent({
+    super.key,
+    super.text,
+    super.leadingIcon,
+    super.trailingIcon,
+    super.iconSize,
+    super.child,
+    super.onPressed,
+    super.onLongPress,
+    ButtonStyle? style,
+    super.size,
+    super.expanded,
+    required BuildContext context,
+  }) : super(
+          style: style == null
+              ? ButtonStyles.transparentTextButtonStyle(
+                  Theme.of(context).primaryColor,
+                )
+              : style.merge(
+                  ButtonStyles.transparentTextButtonStyle(
+                    Theme.of(context).primaryColor,
+                  ),
```

Comment:
@Shakle can u have a look at how would u do this?
Not passing the required build context there, same for the `.hint` button, from line 65-71, i also pass the context

Reply:
@cgutierr-zgz created a flow to solve it in 'buttons' branch. So it works, but it's a bit more complicated as it initially was :D But the constructor is clear now :)